src/intro_movie.c

```diff
@@ -0,0 +1,197 @@
+#include "global.h"
+#include "intro_movie.h"
+#include "intro_movie_internal.h"
+#include "constants/sndseq.h"
+#include "bg_window.h"
+#include "brightness.h"
+#include "gf_gfx_planes.h"
+#include "main.h"
+#include "math_util.h"
+#include "sound_02004A44.h"
+#include "sys_task_api.h"
+#include "system.h"
+#include "title_screen.h"
+#include "unk_0200B150.h"
+#include "unk_0200FA24.h"
+#include "unk_020215A0.h"
+#include "unk_02022588.h"
+#include "unk_02023694.h"
+
+BOOL IntroMovie_Init(OVY_MANAGER *man, int *state);
+BOOL IntroMovie_Main(OVY_MANAGER *man, int *state);
+BOOL IntroMovie_Exit(OVY_MANAGER *man, int *state);
+void IntroMovie_SetGraphicsBanks(void);
+void IntroMovie_HandleSpriteUpdates(IntroMovieOvyData *data);
+void IntroMovie_InitSpriteGraphicsHW(IntroMovieOvyData *data);
+void IntroMovie_TeardownSpritesManager(IntroMovieOvyData *data);
+
+const OVY_MGR_TEMPLATE gApplication_IntroMovie = {IntroMovie_Init, IntroMovie_Main, IntroMovie_Exit, FS_OVERLAY_ID_NONE};
+
+static BOOL (*sIntroMovieSceneFuncs[])(IntroMovieOvyData *data, void *a1) = {
+    IntroMovie_Scene1,
+    IntroMovie_Scene2,
+    IntroMovie_Scene3,
+    IntroMovie_Scene4,
+    IntroMovie_Scene5,
+};
+
+HeapID _deadstrip_00(int idx);
+HeapID _deadstrip_00(int idx) {
+    static const HeapID sDeadstrippedRodata[2] = {HEAP_ID_INTRO_MOVIE, HEAP_ID_INTRO_MOVIE};
+    return sDeadstrippedRodata[idx];
+}
+
+BOOL IntroMovie_Init(OVY_MANAGER *man, int *state) {
+    ScreenBrightnessData_InitAll();
+    sub_0200FBF4(PM_LCD_TOP, RGB_BLACK);
+    sub_0200FBF4(PM_LCD_BOTTOM, RGB_BLACK);
+    Main_SetVBlankIntrCB(NULL, NULL);
+    Main_SetHBlankIntrCB(NULL, NULL);
+    GfGfx_DisableEngineAPlanes();
+    GfGfx_DisableEngineBPlanes();
+    SetKeyRepeatTimers(4, 8);
+    CreateHeap(HEAP_ID_3, HEAP_ID_INTRO_MOVIE, 0x68000);
+
+    IntroMovieOvyData *data = OverlayManager_CreateAndGetData(man, sizeof(IntroMovieOvyData), HEAP_ID_INTRO_MOVIE);
+    memset(data, 0, sizeof(IntroMovieOvyData));
+    data->heapID = HEAP_ID_INTRO_MOVIE;
+    data->introSkipped = FALSE;
+    data->skipAllowed = FALSE;
+    gSystem.screensFlipped = TRUE;
+    GfGfx_SwapDisplay();
+    data->savedLCRngSeed = GetLCRNGSeed();
+    SetLCRNGSeed(0);
+    data->bgConfig = BgConfig_Alloc(data->heapID);
+    IntroMovie_SetGraphicsBanks();
+    IntroMovie_InitSpriteGraphicsHW(data);
+    return TRUE;
+}
+
+BOOL IntroMovie_Main(OVY_MANAGER *man, int *state) {
+    IntroMovieOvyData *data = OverlayManager_GetData(man);
+    if (data->skipAllowed && ((gSystem.newKeys & PAD_BUTTON_A) || (gSystem.newKeys & PAD_BUTTON_START) || gSystem.touchNew)) {
+        data->introSkipped = TRUE;
+        gSystem.unk70 = FALSE;
+        sub_0200FBF4(PM_LCD_TOP, RGB_WHITE);
```

Comment:
this function fills the designated screen with that one colour right?

Reply:
ie it can probably be named

Reply:
it's not so simple

Reply:
besides which, it's out of scope for this PR